packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -34,6 +35,138 @@ typedef ScrollableWidgetBuilder = Widget Function(
+///
+/// The controller's methods cannot be used until after the controller has been
+/// passed into a [DraggableScrollableSheet] and the sheet has run initState.
+class DraggableScrollableController {
+  _DraggableScrollableSheetScrollController? _attachedController;
+
+  /// Get the current size (as a fraction of the parent height) of the attached sheet.
+  double get size {
+    _assertAttached();
+    return _attachedController!.extent.currentSize;
+  }
+
+  /// Get the current pixel height of the attached sheet.
+  double get pixels {
+    _assertAttached();
+    return _attachedController!.extent.currentPixels;
+  }
+
+  /// Convert a sheet's size (fractional value of parent container height) to pixels.
+  double sizeToPixels(double size) {
+    _assertAttached();
+    return _attachedController!.extent.sizeToPixels(size);
+  }
+
+  /// Convert a sheet's pixel height to size (fractional value of parent container height).
+  double pixelsToSize(double pixels) {
+    _assertAttached();
+    return _attachedController!.extent.pixelsToSize(pixels);
+  }
+
+  /// Animates the attached sheet from its current size to [size] to the
+  /// provided new `size`, a fractional value of the parent container's height.
+  ///
+  /// Any active sheet animation is canceled. If the sheet's internal scrollable
+  /// is currently animating (e.g. responding to a user fling), that animation is
+  /// canceled as well.
+  ///
+  /// An animation will be interrupted whenever the user attempts to scroll
+  /// manually, whenever another activity is started, or when the sheet hits its
+  /// max or min size (e.g. if you animate to 1 but the max size is .8, the
+  /// animation will stop playing when it reaches .8).
+  ///
+  /// The duration must not be zero. To jump to a particular value without an
+  /// animation, use [jumpTo].
+  ///
+  /// When calling [animateTo] in widget tests, `await`ing the returned
+  /// [Future] may cause the test to hang and timeout. Instead, use
+  /// [WidgetTester.pumpAndSettle].
+  Future<void> animateTo(
+    double size, {
+    required Duration duration,
+    required Curve curve,
+  }) async {
+    _assertAttached();
+    assert(size >= 0 && size <= 1);
+    assert(duration != Duration.zero);
+    final AnimationController animationController = AnimationController.unbounded(
+      vsync: _attachedController!.position.context.vsync,
+      value: _attachedController!.extent.currentSize,
+    );
+    _attachedController!.position.goIdle();
+    // This disables any snapping until the next user interaction with the sheet.
+    _attachedController!.extent.hasDragged = false;
+    _attachedController!.extent.startActivity(onCanceled: () {
+      // Don't stop the controller if it's already finished and may have been disposed.
+      if (animationController.isAnimating) {
+        animationController.stop();
+      }
+    });
+    CurvedAnimation(parent: animationController, curve: curve).addListener(() {
```

Comment:
The value of the `CurvedAnimation` is never used, so the curve is never applied to the animation.

Reply:
The curved animation is referenced via `.addListener`

Reply:
Thank you for this PR, it's a fortunate timing as we need this feature in our project :)

We tested it on our project, and noticed the animation was always linear.  

Maybe I'm missing something but I don't think `CurvedAnimation` works this way.  
`.addListener` just forwards the ticker subscription to the animation parent (`animationController`).
The actual curved value is never used. The value used in the callback is `animationController.value` (which is linear by default).  
Using the value of the `CurvedAnimation` has some issue though: as the animation does not always start at `0.0`, it would jump at the beginning.  You could fix this with a [0,1] `AnimationController` and a `Tween`.  

But as `animateTo` supports a curve argument, a simple fix would be:
```dart
animationController.addListener(() {
  _attachedController!.extent.updateSize(
    animationController.value,
    _attachedController!.position.context.notificationContext!,
  );
  if (animationController.value > _attachedController!.extent.maxSize ||
      animationController.value < _attachedController!.extent.minSize) {
    // Animation hit the max or min size, stop animating.
    animationController.stop(canceled: false);
  }
});
await animationController.animateTo(size, duration: duration, curve: curve);
```

Reply:
I'll make another PR if it's ok with you, as yours is already merged.

Reply:
ooh good catch, sorry for dismissing it at first.

Feel free to make a new PR, though make sure you also introduce a test that fails before the fix and succeeds after it-it was sloppy on my part to not test the curve argument in the first place. `dnfield` is probably the right person to review your PR, though `goderbauer` might want to look at it as well (not `@`'ing them here so as not to spam them).

Reply:
Just got around to creating a PR:
https://github.com/flutter/flutter/pull/95744

Sorry for the delay, had a lot of other stuff going on.

Reply:
Thank you for the fix! Did not had much time to handle the PR last few weeks either :(

Reply:
Looks like the PR should be going in soon, hopefully it'll make it into the next Flutter release.

Let me know if there are any other things you run into while trying to use `DraggableScrollableSheet`!